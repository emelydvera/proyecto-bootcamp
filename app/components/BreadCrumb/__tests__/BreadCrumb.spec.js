const React = require('react');
const { render, screen } = require('@testing-library/react')
const BreadCrumb = require('..');

const path = [
  {
    "id": "MLA1648",
    "name": "ComputaciÃ³n"
  },
  {
    "id": "MLA430687",
    "name": "Laptops y Accesorios"
  },
]

describe('BreadCrumb', () => {
  it('Should render correctly', () => {

    const { asFragment } = render(<BreadCrumb path={path} />)
    expect(asFragment()).toMatchSnapshot();

  })

  it('Should render the breadcrumb levels, with it custom href', () => {
    render(<BreadCrumb path={path} />)
    const levelOne = screen.queryByText(path[0].name);
    expect(levelOne).toBeInTheDocument()
    expect(levelOne).toHaveAttribute('href', expect.stringContaining(path[0].id))
    screen.debug(levelOne)
  })

  it('Should add the product title into the breadcrumb levels', () => {
    render(<BreadCrumb path={path} productTitle={'Macbook'} />)
    const productTitle = screen.queryByText(/Macbook/);
    expect(productTitle).toBeInTheDocument()
  })


})