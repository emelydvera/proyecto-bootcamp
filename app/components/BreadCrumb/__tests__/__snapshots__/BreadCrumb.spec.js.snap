// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb Should render correctly 1`] = `
<DocumentFragment>
  <nav
    aria-label="lista de páginas"
  >
    <ol
      class="andes-breadcrumb breadcrumb"
    >
      <li
        class="andes-breadcrumb__item"
      >
        <a
          class="andes-breadcrumb__link"
          href="/listado?category=MLA1648"
          title="Ir a la sección Computación"
        >
          Computación
        </a>
        <div
          class="andes-breadcrumb__chevron"
        >
          <svg
            height="8"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 0l4 4-4 4"
              fill="none"
              stroke="#666"
            />
          </svg>
        </div>
      </li>
      <li
        class="andes-breadcrumb__item"
      >
        <a
          aria-current="page"
          class="andes-breadcrumb__link"
          href="/listado?category=MLA430687"
          title="Ir a la sección Laptops y Accesorios"
        >
          Laptops y Accesorios
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
